lib/routes/openstack.js

```diff
@@ -0,0 +1,211 @@
+ * under the License.
+ */
+'use strict';
+
+var http = require('http');
+
+
+/**
+ * @function parseRegionNameFromUserName
+ * Get region name from username. username MUST be admin-regionName
+ * @param {String} username
+ */
+function parseRegionNameFromUserName(username) {
+
+    var  index = username.indexOf('-');
+    if (index <= 0) {
+        throw new Error ('invalid username ' + username);
+    }
+    if ( username.indexOf('@') >= 0 ) {
+        throw new Error ('invalid username %s. Username can not be an email', username);
+    }
+    return username.substring(index + 1);
+}
+
+
+/**
+ * @function getValidateToken
+ * Call to keystone and validate token.
+ * @param {String} adminToken
+ * @param {String} token
+ * @param {function} callback
+ */
+function getValidateToken(adminToken, token, callback) {
+    var headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'X-Auth-Token': adminToken,
+        'X-Subject-Token': token
+    };
+
+    var options = {
+        host: 'cloud.lab.fiware.org',
+        port: '4731',
+        path: '/v3/auth/tokens',
+        method: 'GET',
+        headers: headers
+    };
+
+    var req = http.request(options, function (res) {
+        res.setEncoding('utf-8');
+        var responseString = '';
+
+        res.on('data', function (data) {
+            responseString += data;
+        });
+        res.on('end', function () {
+            console.log('Response string: %s', responseString);
+            try {
+                var resultObject = JSON.parse(responseString);
+                var region = parseRegionNameFromUserName(resultObject.token.user.name);
+                callback(region);
+            } catch (ex) {
+                console.log('Error in parse response string:  %s %s', responseString, ex);
+                callback();
+            }
+        });
+
+    });
+    req.on('error', function (e) {
+        if (e.code === 'ECONNRESET') {
+            console.log('Error in connection by TIMEOUT with keystone: ' + e);
+        } else {
+            console.log('Error in connection with keystone: ' + e);
+        }
+        callback();
+
+
+    });
+
+    req.setTimeout(10000, function () {
+        req.abort();
+        console.log('Timeout in the connection with keystone. Time exceed');
+    });
+
+
+    req.end();
+
+
+}
+
+/**
+ * @function postAdminToken
+ * Call to keystone and get new admin token.
+ * @param {function} callback
+ */
+function postAdminToken(callback) {
+
+    var payload = {
+        auth: {
+            identity: {
+                methods: [
+                    'password'
+                ],
+                password: {
+                    user: {
+                        domain: {
+                            id: 'default'
+                         },
+                        name: 'admin',
+                        password: 'asXK3r9V4n1Ad93'
```

Comment:
It should not be like a configuration parameter in order that you cannot show this "sensible" data in the repository?